### ISSUE TITLE
Re-run after modifying config file

### ISSUE THREAD

**Issue by hcpl** [rust-lang/rust-clippy#2130]
For cases if after adding/removing rules:
- `clippy` behaves differently on the same source;
- config contains invalid keys or values.

Currently, if the run before modifying config was successful then the run after that will silently pass too which can give a false assertion "your code is fine even now". It would be useful to trigger `clippy` re-run without having to `touch` the source itself.

To reproduce the 2nd use-case:
```sh
$ cat clippy.toml
enum-variant-name-threshold = 0
$ cargo +nightly clippy
# run successful
$ echo 'foo = "bar"' >> clippy.toml
$ cargo +nightly clippy
# run successful but should fail with
# "error reading Clippy's configuration file: unknown field `foo`"
$ touch src/main.rs  # whatever source file
$ cargo +nightly clippy
# fails with "error reading Clippy's configuration file: unknown field `foo`"
# ^ expected behavior on the previous run
```
